src/cargo/ops/cargo_update.rs

```diff
@@ -637,15 +641,17 @@ fn print_lockfile_updates(
         };
 
         if let Some((removed, added)) = diff.change() {
-            let latest = report_latest(&possibilities, *added).unwrap_or_default();
+            let required_rust_version = report_required_rust_version(ws, resolve, *added);
+            let latest = report_latest(&possibilities, *added);
+            let note = required_rust_version.or(latest).unwrap_or_default();
 
             let msg = if removed.source_id().is_git() {
                 format!(
-                    "{removed} -> #{}",
+                    "{removed} -> #{}{note}",
                     &added.source_id().precise_git_fragment().unwrap()[..8],
                 )
             } else {
-                format!("{removed} -> v{}{latest}", added.version())
+                format!("{removed} -> v{}{note}", added.version())
```

Comment:
How does it looks like when removing a package required incompatible rust version? It seems that we don't have such a test.

Reply:
This case isn't actually removing but "Updating" or "Downgrading" and we are reporting the status of the package we are upgrading or downgrading to.

Pulled from our tests
```
-[UPDATING] bar v1.5.0 -> v1.6.0
+[UPDATING] bar v1.5.0 -> v1.6.0 (requires Rust 1.65.0)
```

A proper remove has no status being reported
```rust
            for package in diff.removed.iter() {
                ws.gctx().shell().status_with_color(
                    "Removing",
                    format!("{package}"),
                    &style::ERROR,
                )?;
            }
```

Reply:
I was confused. Checked this branch again and yes it is not "removed". Thanks for the explanation.